plugins/parsers/json/parser_test.go

```diff
@@ -948,3 +970,304 @@ func TestParse(t *testing.T) {
 		})
 	}
 }
+
+func TestParseWithWildcardTagKeys(t *testing.T) {
+	// Test that wildcard matching with tags nested within object works
+	parser, err := New(&Config{
+		MetricName: "json_test",
+		TagKeys:    []string{"tags_object_*"},
+	})
+	require.NoError(t, err)
+	metrics, err := parser.Parse([]byte(validJSONTags))
+	require.NoError(t, err)
+	require.Len(t, metrics, 1)
+	require.Equal(t, "json_test", metrics[0].Name())
+	require.Equal(t, map[string]interface{}{
+		"a":   float64(5),
+		"b_c": float64(6),
+	}, metrics[0].Fields())
+	require.Equal(t, map[string]string{
+		"tags_object_mytag":    "foobar",
+		"tags_object_othertag": "baz",
+	}, metrics[0].Tags())
```

Comment:
Please change this to a config-expected table approach. That is, define a new struct e.g.
```
var tests = []struct {
		config       []string
		expected  telegraf.Metric
	}{
		[]string{"tags_object_*"},
		testutil.MustMetric(
			"json_test",
			map[string]string{
				"tags_object_mytag":    "foobar",
				"tags_object_othertag": "baz",
			},
			map[string]interface{}{
￼				"a":   float64(5),
￼		￼		"b_c": float64(6),
￼			},
			time.Unix(0, 0),
		)
	},
	{
	...
	},
```
and then iterate over this table running your tests. You can even compare the resulting metrics using`testutil.RequireMetricEqual()`.

Reply:
I've refactored to using the config-expected table approach :)